refactor: update slack api

guyfedwards opened the pull request.

guyfedwards committed:
build: bump runtime to 10.x

guyfedwards committed:
refactor: update slack apis

rethrow some errors with logging

benswinburne commented on src/slack.js:
```diff
@@ -9,16 +9,24 @@ const instance = new Slack(process.env.ACCESS_TOKEN);
-    .promise();
+  let res;
+  try {
+    res = await s3
```
Why not just return here rather than assigning it?

guyfedwards replied:
leftover from debugging

benswinburne commented on serverless.yml:
```diff
@@ -8,7 +8,7 @@ plugins:
   name: aws
   stage: dev
-  runtime: nodejs8.10
+  runtime: nodejs10.x
```
Why not 12 or 14?

guyfedwards replied:
because 10 worked and 16 didn't and I couldn't be bothered to try in between

benswinburne replied:
well node 16 doesn't exist on lambda so that was always a bit of a non starter...

12 or 14 have much faster cold start times so might help alleviate the problem where slack retries it a few times when it doesn't immediately respond and we get a few attempts at the emoji?

guyfedwards replied:
14 doesn't seem to hang on response like 16 so have updated to that

guyfedwards committed:
ci: use nodejs10.x circle

benswinburne reviewed (changes requested):
iirc there was a reason we didn't throw e everywhere but i don't remember why, do you?

guyfedwards committed:
build: update to node 14.x

guyfedwards commented:
> iirc there was a reason we didn't throw e everywhere but i don't remember why, do you?

No idea, this was causing slack to keep retrying on error which is why I'm now throwing to 500 the handler. To be honest the whole error handling situation could do with some love

benswinburne commented:
> > iirc there was a reason we didn't throw e everywhere but i don't remember why, do you?
> 
> No idea, this was causing slack to keep retrying on error which is why I'm now throwing to 500 the handler. To be honest the whole error handling situation could do with some love

https://github.com/guyfedwards/emojo/issues/24 :D

benswinburne commented on .circleci/config.yml:
```diff
@@ -9,7 +9,7 @@ jobs:
 
     docker:
-      - image: lambci/lambda:build-nodejs8.10
+      - image: lambci/lambda:build-nodejs10.x
```
Needs to match 14 of lambda runtime

guyfedwards committed:
ci: use npm to install yarn

yum threw a wobbly because it hadn't been used to install nodejs
and so installing yarn failed because of missind dependency nodejs

benswinburne committed:
ci: added checkout command
